test-core-fe/src/data/mock-data-provider.ts

```diff
@@ -32,15 +33,114 @@ const comments = (await mockComments({count: 50})).map((comment) => {
   }
 })
 
+const versions = (await mockVersions({ count: 20 })).map((version) => {
+  const user = faker.helpers.arrayElement(users)
+  const resource = faker.helpers.arrayElement([
+    faker.helpers.arrayElement(users),
+    faker.helpers.arrayElement(companies),
+  ])
+
+  return {
+    ...version,
+
+    actor_id: user.id,
+    actor_type: user.type,
+
+    resource_id: resource.id,
+    resource_type: resource.type,
+
+    // author: user,
+    // resource: resource
+  }
+})
+
+//a transfer is 2 lines, so 8 mock transfers returns 16 lines
+const lines = (await mockTransfers({ count: 150 })).map((line, idx) => {
+  console.log("asdf")
+  const user = faker.helpers.arrayElement(users)
+  const resource = faker.helpers.arrayElement([
+    faker.helpers.arrayElement(users),
+    faker.helpers.arrayElement(companies),
+  ])
+  let scope, partnerScope
+  if (!(idx % 2)) { //line 1
```

Comment:
this logic won't create lines with matching scope & partner scope. you are regenerating a new user and resource with each pass through the line array (not by pair of lines).